### PR TITLE
[DSM] PEPPER-212 Dashboard Counts Table View

### PR DESCRIPTION
**Pull request by GiCharkviani**
[PEPPER-212](https://broadworkbench.atlassian.net/browse/PEPPER-212)

Created table view for counts part of Statistics Dashboard

Screenshot 2022-10-27 at 15 08 52

**Functionalities:**
1. Filter data either by title or count
2. Sort data either by title or count

**Unit Tests:**
Test cases include:
- Testing for component creation
- Testing for filter
- Testing for sort
- Testing for all data
- Testing for no data

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/cards-table/counts-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  Component,
+  Input,
+  ViewChild
+} from '@angular/core';
+import {CountsModel} from '../../models/Counts.model';
+import {MatTableDataSource} from '@angular/material/table';
+import {MatSort} from '@angular/material/sort';
+
+@Component({
+  selector: 'app-cards-table',
```
Comment (SimoneMaiwald) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/cards-table/counts-table.component.ts]: Could we change the selector to `app-counts-table` having cards table here is a bit confusing. maybe also renaming the component folder from `cards-table` to `counts-table`

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/cards-table/counts-table.component.ts]: Fixed

File: ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/cards-table/counts-table.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  /**
+   * Checks after filtering by non-existing title/count, if there is displayed only one row with appropriate context
+   */
+  it('should filter zero item', async () => {
+    const NOT_FOUND = 'notFound';
+    const matRowHarnesses: MatRowHarness[] = await getTableRowsAfterFilter(NOT_FOUND);
+    const [firstCell]: MatCellHarness[] = await matRowHarnesses[0].getCells();
+    const firstCellText: string = await firstCell.getText();
+
+    expect(firstCellText).toBe('No data matching the filter ' + '"' + NOT_FOUND + '"');
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/cards-table/counts-table.component.spec.ts]: **_#optional_**

Template literals with backticks ( ` ) can be used here as a short option.

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/cards-table/counts-table.component.spec.ts]: Oh, forgot to fix that. Fixed!

File: ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/dashboard-statistics.component.ts [hunk 3]
```diff
@@ -13,7 +14,7 @@ import {HttpErrorResponse} from '@angular/common/http';
 
 export class DashboardStatisticsComponent implements OnInit {
   Charts: Observable<any>;
-  Counts: Observable<any>;
+  Counts: Observable<CountsModel[]>;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/dashboard-statistics.component.ts:17]: Is it okay that you pass Observable to `MatTableDataSource`?

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/dashboard-statistics.component.ts:17]: **This observable is subscribed by async pipe in the template and CountsModel is passed in the end,** 

but the parent component and service implementations (I mean dashboard-statistics component and service) require further improvements and I will take care of that next time.

Comment (alextanlan) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/dashboard-statistics.component.ts:17]: >  This observable is subscribed by async pipe in the template 

I see, it was done previously. So, I missed it.

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/dashboard-statistics.component.ts:17]: It's okay. Anyway, I have to change things there later. I don't like how it looks, also + some unit tests for them

**Review by alextanlan — APPROVED** [review]

**Pull request merged**